### PR TITLE
Warn user when generic plugin is detected in diagnose

### PR DESCRIPTION
**Pull request by sridhargaddam**
Although "generic" network plugin works with many CNIs,
there is no guarantee that it will work with all.
So, instead of reporting success, this PR displays
appropriate warning.

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

**Review by Jaanki — APPROVED** [review]

File: pkg/subctl/cmd/diagnose/cni.go [hunk 1]
```diff
@@ -82,7 +82,12 @@ func checkCNIConfig(cluster *cmd.Cluster) bool {
 		return false
 	}
 
-	status.EndWithSuccess("The detected CNI network plugin (%q) is supported", cluster.Submariner.Status.NetworkPlugin)
+	if cluster.Submariner.Status.NetworkPlugin == constants.NetworkPluginGeneric {
```
Comment (maayanf24) [inline pkg/subctl/cmd/diagnose/cni.go:85]: I think in case of Generic network plugin, the code will return false on L82 and will not reach this line

Comment (sridhargaddam) [reply pkg/subctl/cmd/diagnose/cni.go:85]: Oops, I accidentally removed the Generic from the list of supported plugins. Good catch @maayanf24

**Review by maayanf24 — APPROVED** [review]

File: pkg/subctl/cmd/diagnose/cni.go [hunk 2]
```diff
@@ -82,7 +82,12 @@ func checkCNIConfig(cluster *cmd.Cluster) bool {
...
+		status.EndWithWarning("Submariner could not detect the CNI network plugin and is using (%q) plugin."+
+			" It may or may-not work.", cluster.Submariner.Status.NetworkPlugin)
```
Comment (nyechiel) [inline pkg/subctl/cmd/diagnose/cni.go]:
```suggestion
			" It may or may not work.", cluster.Submariner.Status.NetworkPlugin)
```

Comment (nyechiel) [reply pkg/subctl/cmd/diagnose/cni.go]: nit: may not

**Review by dfarrell07 — APPROVED** [review]

**Comment by nyechiel** [comment]
@skitt can you please review?

**Review by skitt — APPROVED** [review]

**Pull request merged**